### ISSUE TITLE
What would be the macro equivalent of build_runner's built.yaml?

### ISSUE THREAD

**Issue by rrousselGit** [dart-lang/language#2220]
Hello!

I just thought about something that I haven't seen discussed before: How would macros offer project-wide configurations?

It's pretty common for existing code-generators to allow enabling/disabling features globally through their `build.yaml` file.  
There are some discussions about configurations here https://github.com/dart-lang/language/blob/master/working/macros/feature-specification.md#environment-variables-and-configuration but it seems to be talking about alternative features like build flags – which I believe is a different topic.

In particular, in the link mentioned before, it mentions:
>  Making macros configuration-free allows tools to give users a single, consistent version of generated code.

but this does not apply to a `build.yaml` like configuration file.

Will macros have something built-in for this?

I suppose we could use the Resource API for this, manually reading a file + decoding it. But I could see this leading to macros parsing the configuration multiple times, making this less efficient than it should be.

**Comment by jakemac53** [comment]
Yeah, I think today you would do this with the Resource API, but it's a good point that you would end up duplicating work.

We need to be able to track the files a given macro application depends on still, so there does need to be an explicit api exposed to support that dependency tracking combined with using a cached result.
